optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/termination/BasicPlumbingTermination.java

```diff
@@ -35,7 +41,8 @@
     protected final boolean daemon;
 
     protected boolean terminatedEarly = false;
-    protected BlockingQueue<ProblemFactChange<Solution_>> problemFactChangeQueue = new LinkedBlockingQueue<>();
+
+    protected BlockingQueue<ProblemChangeAdapter<Solution_>> problemFactChangeQueue = new LinkedBlockingQueue<>();
```

Comment:
Do we want to support both old and new PFCs on the same `Solver` instance?
If yes, they should both end up in the same queue and thus, we need the `ProblemChangeAdapter`.